### PR TITLE
[Windows] Add checkbox widget

### PR DESCRIPTION
**Pull request by citycide**
I'm not very familiar with the Windows API yet (and just getting started in low-level code in general) so this is mostly a modification of the existing controls. Comments definitely welcome.

### CODE REVIEW

File: src/nigui/private/windows/platform_impl.nim [hunk 1]
```diff
@@ -1216,6 +1216,57 @@ method `enabled=`(button: NativeButton, enabled: bool) =
...
+  case uMsg
+  of WM_KEYDOWN, WM_LBUTTONUP:
+    let checkbox = cast[Checkbox](pGetWindowLongPtr(hWnd, GWLP_USERDATA))
+    let keyCode = if uMsg == WM_KEYDOWN: cast[int](wParam) else: 32
+    if checkbox != nil and keyCode == 32:
+      checkbox.fChecked = not checkbox.fChecked
+      discard SendMessageA(checkbox.fHandle, BM_SETCHECK, cast[pointer](checkbox.fChecked), nil)
+
+      var event = new CheckboxToggleEvent
+      event.control = checkbox
```
Comment (citycide) [inline src/nigui/private/windows/platform_impl.nim]: At this point the checkbox is cast to a generic `Control` which makes using `event.control` a bit messy, I think you have to recast it back to a `Checkbox`. Would like input on that

File: src/nigui/private/windows/windows.nim [hunk 2]
```diff
@@ -35,8 +35,15 @@ const
   ACTCTX_FLAG_RESOURCE_NAME_VALID* = 8
   ACTCTX_FLAG_SET_PROCESS_DEFAULT* = 16
   BN_CLICKED* = 0
+  BM_GETCHECK* = 240.int32
+  BM_SETCHECK* = 241.int32
```
Comment (citycide) [inline src/nigui/private/windows/windows.nim]: Without these set to `int32` building fails when compiling on a Win x64 machine, and probably extends to other areas too but I'm not sure the best way to go about this yet.

File: src/nigui/private/windows/windows.nim [hunk 3]
```diff
@@ -127,10 +134,13 @@ const
   WS_EX_CLIENTEDGE* = 0x00000200
   WS_GROUP* = 0x00020000
   WS_HSCROLL* = 0x00100000
+  WS_MAXIMIZEBOX* = 0x00010000
+  WS_MINIMIZEBOX* = 0x00020000
```
Comment (citycide) [inline src/nigui/private/windows/windows.nim]: Some of these new constants aren't really relevant for the checkbox but I figured they'll end up getting used for things like non-resizable windows. That's what I used these for but wasn't sure how to implement them as an option for a window - I used it for all new windows in my local version.

**Comment by trustable-code** [comment]
After long time, the checkboxes are now implemented :)

**Pull request closed without merging**